worker/src/RTC/DtlsTransport.cpp

```diff
@@ -1022,9 +1047,24 @@ namespace RTC
 			MS_WARN_TAG(
 			  dtls, "OpenSSL SSL_write() wrote less (%d bytes) than given data (%zu bytes)", written, len);
 		}
+	}
+
+	void DtlsTransport::SendDtlsData(const uint8_t* data, size_t len)
+	{
+		MS_TRACE();
+
+		MS_DEBUG_DEV("%zu bytes of DTLS data ready to be sent", len);
 
-		// Send data.
-		SendPendingOutgoingDtlsData();
+		// Notify the listener.
+		this->listener->OnDtlsTransportSendData(this, data, len);
+
+		// Clear the BIO buffer.
+		// NOTE: Here we now that OpenSSL called the onSslBioOut() callback with
+		// with all the byte length that it previously wrote into the BIO mem
+		// buffer (this->sslBioToNetwork), so it's safe to clear the entire buffer
+		// after sending its entire DTLS data to the endpoint.
+		// NOTE: the (void) avoids the -Wunused-value warning.
+		(void)BIO_reset(this->sslBioToNetwork);
```

Comment:
`BIO_reset()` returns -1 on failure. It would be nice to log it in case of error.

Reply:
It must return 1 on success on our BIO. Pushed.